src/components/PaginationWithRear/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { ReactNode } from 'react'
+import classNames from 'classnames'
+import Pagination from '../Pagination'
+import styles from './styles.module.scss'
+
+const PaginationWithRear = ({
+  currentPage,
+  totalPages,
+  onChange,
+  paginationClassName,
+  rear,
+}: {
+  currentPage: number
+  totalPages: number
+  onChange: (page: number) => void
+  paginationClassName?: string
+  rear: ReactNode
+}) => {
+  return (
+    <div className={classNames(styles.paginationWithRear, { [styles.withPagination]: totalPages > 1 })}>
+      {totalPages > 1 && (
+        <Pagination
+          currentPage={currentPage}
+          totalPages={totalPages}
+          onChange={onChange}
+          className={classNames(styles.pagination, paginationClassName)}
+        />
+      )}
+      <div className={styles.rear}>{rear}</div>
+    </div>
+  )
+}
+
+export default PaginationWithRear
```

Comment:
I would prefer `export default` if there's only one module to export. And using `export default` forces developers to make modules small and clear.

Reply:
I have re-examined the convention of "avoid using import/export default as much as possible", which was originally established for the following reasons:

1. When enabling/disabling `esModuleInterop` and `allowSyntheticDefaultImports`, there have been compatibility issues with older module systems, but I can't recall any specific examples at the moment, so this factor can be disregarded for now.
2. Previously, most language server protocols (LSP) did not fully support `default` syntax, such as finding references and renaming (which are external identifiers for default), but from my recent testing, modern LSPs have generally improved in supporting these capabilities.

In modern IDEs, the issues that still exist with `export default` are:

1. The syntax `export default { ... }` can affect tree shaking (not sure if this is still an issue, but it might be?).
2. `export default` without an identifier still cannot be renamed.
3. When manually importing, there is no name suggestion, and it is possible to misspell the name (which affects LSP's rename behavior), or intentionally use a different (abbreviated) identifier. This may sound clean, but we have all seen people importing functions with non-descriptive names (like `import a from './getName'`).

So my conclusion now is:
It is still recommended to use named exports as much as possible, but default exports can also be used.
However, the `export default { ... }` syntax should not be used, and an identifier should be included when using default exports.
When importing the contents of a default export, it is recommended to use the IDE's automatic import feature to avoid misspelling the identifier.